contracts-available/FlightDelayStaking.sol

```diff
@@ -32,230 +48,245 @@ contract FlightDelayRiskPool is Ownable {
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev Set staking ratio
+     * @param _relation is the dip token amount
+     * @param _divider is the stable amount
      */
-    IERC20 public DipTokenContract;
+    function setStakingRelation(uint256 _relation, uint256 _divider)
+        public
+        onlyOwner
+    {
+        stakingRelation.val = _relation;
+        stakingRelation.div = _divider;
+    }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev Set staking ratio
+     * @param _exposureFactor is the new exposureFactor
      */
-    constructor (address _DipContractAddress) {
-        DipTokenContract = DipToken(_DipContractAddress);
-        setStakingRelation(1,1);
+    function setExposureFactor(uint256 _exposureFactor) public onlyOwner {
+        exposureFactor = _exposureFactor;
     }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev Set staking ratio
+     * @param _collatFactor is the new collatFactor
      */
-    function setStakingRelation (uint256 _relation, uint256 _divider) public {
-        stakingRelation.val = _relation;
-        stakingRelation.div = _divider;
+    function setCollatFactor(uint256 _collatFactor) public onlyOwner {
+        collatFactor = _collatFactor;
     }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
+     * @dev Stake stable and dip tokens
+     * @param _stake is the amount of dip token
      *
      */
     function stake(uint256 _stake) public payable {
-
-        stake currentStakersStake = stakeBalances[msg.sender];
-        increaseCurrentStake(_stake);
-        uint256 totalStableStake = currentSingleStake.stable + msg.value;
-        uint256 totalDipStake = currentStake.dip + _stake;
+        Stake memory currentStakersStake = stakeBalances[msg.sender];
+        uint256 totalStableStake = currentStakersStake.stable + msg.value;
+        uint256 totalDipStake = currentStakersStake.dip + _stake;
         uint256 requiredDip = calculateRequiredDip(totalStableStake);
-        require (totalDipStake <= requiredDip, "Stake to high");
-        require (DipTokenContract.transferFrom(msg.sender, this, stake), "DIP could not be staked");
-        stakeBalances[msg.sender] = stake(totalStableStake, totalDipStake);
+        require(totalDipStake <= requiredDip, "Stake to high");
```

Comment:
replace <= by == because it does not make sense to stake more DIP then necessary